lib/trino-parquet/src/main/java/io/trino/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -106,11 +110,31 @@ public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor
                 return createDecimalColumnReader(descriptor).orElse(new BinaryColumnReader(descriptor));
             case FIXED_LEN_BYTE_ARRAY:
                 return createDecimalColumnReader(descriptor)
-                        .orElseThrow(() -> new TrinoException(NOT_SUPPORTED, " type FIXED_LEN_BYTE_ARRAY supported as DECIMAL; got " + descriptor.getPrimitiveType().getOriginalType()));
+                        .orElseGet(() -> {
+                            // Iceberg 0.11.1 writes UUID as FIXED_LEN_BYTE_ARRAY without logical type annotation (see https://github.com/apache/iceberg/pull/2913)
+                            if (isLogicalUuid(descriptor.getPrimitiveType()) || descriptor.getPrimitiveType().getLogicalTypeAnnotation() == null) {
+                                return new UuidColumnReader(descriptor);
+                            }
+                            throw new TrinoException(NOT_SUPPORTED, " type FIXED_LEN_BYTE_ARRAY supported as DECIMAL; got " + descriptor.getPrimitiveType().getOriginalType());
```

Comment:
The start of this message seems to be missing something.

Grammar nit: don't use "got" as that's informal.

> Non-annotated Parquet type FIXED_LEN_BYTE_ARRAY expected to be DECIMAL, not %s

Reply:
Ah, I see this is existing, but we should fix it

Reply:
fixed